src/POData/UriProcessor/QueryProcessor/QueryProcessor.php

```diff
@@ -510,15 +510,28 @@ private function readSkipOrTopOption($queryItem, &$value)
     private function checkForEmptyQueryArguments()
     {
         $serviceHost = $this->service->getHost();
-        if (!is_null($serviceHost->getQueryStringItem(ODataConstants::HTTPQUERY_STRING_FILTER))
-            || !is_null($serviceHost->getQueryStringItem(ODataConstants::HTTPQUERY_STRING_EXPAND))
-            || !is_null($serviceHost->getQueryStringItem(ODataConstants::HTTPQUERY_STRING_INLINECOUNT))
-            || !is_null($serviceHost->getQueryStringItem(ODataConstants::HTTPQUERY_STRING_ORDERBY))
-            || !is_null($serviceHost->getQueryStringItem(ODataConstants::HTTPQUERY_STRING_SELECT))
-            || !is_null($serviceHost->getQueryStringItem(ODataConstants::HTTPQUERY_STRING_SKIP))
-            || !is_null($serviceHost->getQueryStringItem(ODataConstants::HTTPQUERY_STRING_SKIPTOKEN))
-            || !is_null($serviceHost->getQueryStringItem(ODataConstants::HTTPQUERY_STRING_TOP))
-        ) {
```

Comment:
Rewritten for simplicity and ease of checking - the foreach loop is harder to stuff up than $MONSTER_CONDITIONAL without detection